include/seqan3/search/algorithm/detail/search_trivial.hpp

```diff
@@ -127,7 +138,7 @@ inline bool search_trivial(cursor_t cur, query_t & query, typename cursor_t::siz
             // Match (when error_left.substitution == 0)
             if (cur.extend_right(query[query_pos]))
             {
-                if (search_trivial<abort_on_hit>(cur, query, query_pos + 1, error_left, delegate) && abort_on_hit)
+                if (search_trivial<abort_on_hit>(cur, query, query_pos + 1, error_left, false, false, delegate) && abort_on_hit)
```

Comment:
I think these ifs can be refactored to one

Reply:
Hm, if I refactor the two if clauses in line 107 and 109 to one:
 if (error_left.substitution == 0 && cur.last_char() == query[query_pos] &&
             search_trivial<abort_on_hit>(cur, query, query_pos + 1, error_left, false, false, delegate) &&
              abort_on_hit)
 the unit test fail, so I am not sure what the compiler does maybe switch order of the single conditions?

Reply:
Mhm, does compiling fail or the unit test itself? If it's the units test then leave it as it is. Maybe the compiler does some weird thing.

Reply:
The unit test itself.

Reply:
Well then we don't change it :)